packages/@aws-cdk/tmp-toolkit-helpers/src/api/toolkit-error.ts

```diff
@@ -61,20 +74,61 @@ export class AuthenticationError extends ToolkitError {
 }
 
 /**
- * Represents an authentication-specific error in the AWS CDK Toolkit.
+ * Represents an error causes by cloud assembly synthesis
+ *
+ * This includes errors thrown during app execution, as well as failing annotations.
  */
 export class AssemblyError extends ToolkitError {
-  constructor(message: string) {
+  /**
+   * An AssemblyError with an original error as cause
+   */
+  public static fromCause(message: string, error: unknown, stacks?: cxapi.CloudFormationStackArtifact[]): AssemblyError {
```

Comment:
Does this need to accept `stacks` as well? I don't see a usage of it and it creates some inconsistent naming with `fromStacks`.

Reply:
Good catch!